airbyte-webapp/src/packages/cloud/services/dbtCloud.ts

```diff
@@ -25,47 +31,35 @@ import { useUpdateWorkspace } from "services/workspaces/WorkspacesService";
 import { useConfig } from "./config";
 
 export interface DbtCloudJob {
-  account: string;
-  job: string;
+  accountId: number;
+  jobId: number;
   operationId?: string;
   jobName?: string;
 }
 export type { DbtCloudJobInfo } from "packages/cloud/lib/domain/dbtCloud/api";
-const dbtCloudDomain = "https://cloud.getdbt.com";
 const webhookConfigName = "dbt cloud";
-const executionBody = `{"cause": "airbyte"}`;
-const jobName = (t: DbtCloudJob) => `${t.account}/${t.job}`;
+const jobName = (t: DbtCloudJob) => `${t.accountId}/${t.jobId}`;
 
 const isDbtWebhookConfig = (webhookConfig: WebhookConfigRead) => !!webhookConfig.name?.includes("dbt");
 
-export const toDbtCloudJob = (operationOrCloudJob: OperationRead | DbtCloudJobInfo): DbtCloudJob => {
-  if ("operationId" in operationOrCloudJob) {
-    const { operationId } = operationOrCloudJob;
-    const { executionUrl } = operationOrCloudJob.operatorConfiguration.webhook || {};
-
-    const matches = (executionUrl || "").match(/\/accounts\/([^/]+)\/jobs\/([^]+)\/run/);
-    if (!matches) {
-      throw new Error(`Cannot extract dbt cloud job params from executionUrl ${executionUrl}`);
-    } else {
-      const [, account, job] = matches;
-
-      return {
-        account,
-        job,
-        operationId,
-      };
-    }
-  } else {
-    const { accountId, jobId, jobName } = operationOrCloudJob;
-    return { account: `${accountId}`, job: `${jobId}`, jobName };
+export const toDbtCloudJob = (operationRead: OperationRead): DbtCloudJob => {
+  if (operationRead.operatorConfiguration.webhook?.webhookType === "dbtCloud") {
+    const dbtCloud = operationRead.operatorConfiguration.webhook.dbtCloud as DbtCloudJob;
+    return {
+      accountId: dbtCloud.accountId,
+      jobId: dbtCloud.jobId,
+    };
   }
+  throw new Error(
+    `Cannot convert operationRead of type ${operationRead.operatorConfiguration.operatorType} to DbtCloudJob`
+  );
```

Comment:
@josephkmh Is this something that can happen in reality, or just a dev catch? In either case, should we log this somehow to monitoring?

Reply:
There isn't really a way to do this more cleverly without maintaining handwritten types for this API: the java code generation doesn't support `anyOf` because java doesn't have proper union types, so we don't get to associate specific enum values with the presence or absence of fields in the auto-generated code.